pkg/cmd/context.go

```diff
@@ -2,49 +2,143 @@ package cmd
 		Long:  "Add/edit API configuration.",
-		Run: func(cmd *cobra.Command, args []string) {
-			level.Info(logger).Log("msg", "api called")
+	}
+
+	var addURL, addName string
+	apiAddCmd := &cobra.Command{
+		Use:   "add",
+		Short: "Add API configuration.",
+		Long:  "Add API configuration.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			return conf.AddAPI(logger, addName, addURL)
 		},
 	}
+
+	apiAddCmd.Flags().StringVar(&addURL, "url", "", "The URL for the Observatorium API.")
+	apiAddCmd.Flags().StringVar(&addName, "name", "", "Provide an optional name to easily refer to the Observatorium Instance.")
+
+	err := apiAddCmd.MarkFlagRequired("url")
+	if err != nil {
+		panic(err)
+	}
+
+	var rmName string
+	apiRmCmd := &cobra.Command{
+		Use:   "rm",
+		Short: "Remove API configuration.",
+		Long:  "Remove API configuration. If only one API is saved, that will be removed. If set to current, current will be set to nil.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			return conf.RemoveAPI(logger, rmName)
+		},
+	}
+
+	apiRmCmd.Flags().StringVar(&rmName, "name", "", "The name of the Observatorium API instance to remove. No need to provide if only one API is saved.")
+
 	switchCmd := &cobra.Command{
 		Use:   "switch",
 		Short: "Switch to another context.",
 		Long:  "View/Add/Edit context configuration.",
-		Run: func(cmd *cobra.Command, args []string) {
-			level.Info(logger).Log("msg", "switch called")
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cntxt := strings.Split(args[0], "/")
+			if len(cntxt) != 2 {
+				return fmt.Errorf("invalid context name: use format <api>/<tenant>")
+			}
+
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			return conf.SetCurrentContext(logger, cntxt[0], cntxt[1])
 		},
 	}
+
 	currentCmd := &cobra.Command{
 		Use:   "current",
 		Short: "View current context configuration.",
 		Long:  "View current context configuration.",
-		Run: func(cmd *cobra.Command, args []string) {
-			level.Info(logger).Log("msg", "current called")
+		RunE: func(cmd *cobra.Command, args []string) error {
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			_, _, err = conf.GetCurrentContext()
+			if err != nil {
+				return err
+			}
+
+			// TODO(saswatamcode): Add flag to display more details. Eg -verbose
+			fmt.Fprintf(os.Stdout, "The current context is: %s/%s\n", conf.Current.API, conf.Current.Tenant)
+			return nil
+		},
+	}
+
+	listCmd := &cobra.Command{
+		Use:   "list",
+		Short: "View all context configuration.",
+		Long:  "View all context configuration.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			conf, err := config.Read(logger)
+			if err != nil {
+				return err
+			}
+
+			for k, v := range conf.APIs {
```

Comment:
If we have API's specified, but no tenants, is that a valid context? It was a bit confusing since only APIs were listed when I had no tenants added - but that means those contexts are not valid / usable, right?

Reply:
Context means an API name and a tenant name using which we can do an operation (will add docs soon to clarify this well 🙂). An API can exist in the config without a tenant (although you can't make requests like that), but a tenant cannot exist without an API.

So in this case, it means they aren't usable. You need to add a tenant under your already added API.

Reply:
I think I understand that, but then we're listing 'unusable' contexts, which is not useful for the user.

Reply:
Hmm, that's correct. Let's add an issue once this is merged and figure out a nice way to display contexts in an easily understandable way in future PRs. Maybe adding a " no tenant, cannot use yet" text beside the API name which has no tenants would be reasonable? 🙂